python/pyarrow/tests/test_array.py

```diff
@@ -250,6 +250,25 @@ def test_array_slice():
             assert arr[start:stop].to_pylist() == arr.to_pylist()[start:stop]
 
 
+def test_array_diff():
+    # ARROW-6252
+    arr1 = pa.array(['foo'])
+    arr2 = pa.array(['foo', 'bar', None])
+    arr3 = pa.array([1, 2, 3])
+    arr4 = pa.array([[], [1], None], type=pa.list_(pa.int64()))
+
+    assert arr1.diff(arr1) == ''
+    assert arr1.diff(arr2) == '''
```

Comment:
The test failure indicates this is comparing the character buffers instead of the arrays. Want me to dig into it?